pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdProduce.java

```diff
@@ -189,6 +208,29 @@ private List<byte[]> generateMessageBodies(List<String> stringMessages, List<Str
         return messageBodies;
     }
 
+    private static byte[] jsonToAvro(String m, org.apache.avro.Schema avroSchema){
+        try {
+            GenericDatumReader<Object> reader = new GenericDatumReader<>(avroSchema);
+            JsonDecoder jsonDecoder = DecoderFactory.get().jsonDecoder(avroSchema, m);
+            GenericDatumWriter<Object> writer = new GenericDatumWriter<>(avroSchema);
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            Encoder e = EncoderFactory.get().binaryEncoder(out, null);
+            Object datum = null;
+            while (true) {
+                try {
+                    datum = reader.read(datum, jsonDecoder);
+                } catch (EOFException eofException) {
+                    break;
+                }
+                writer.write(datum, e);
+                e.flush();
+            }
+            return out.toByteArray();
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot convert " + m + " to AVRO " + e.getMessage(), e);
```

Comment:
since we use JSON decoder, can you improve the error message saying that you need to pass a valid JSON ?

Reply:
The error is already quite informative, because it is the error reported by Jackson